external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -91,11 +91,34 @@ private[avro] case class AvroDataToCatalyst(
         null
     }
 
+  /**
+   * This computes the offset for avro binary decoding based on the presence of a magic number
+   * defined in avro BinaryMessageEncoder.java as of now we are assuming that the binary
+   * provided was encoded via avro BinaryEncoder.java. However some serializers use
+   * the former strategy instead which prepend the binary with a magic number followed
+   * by a 8-byte schema fingerprint. This method detects the presence
+   * of the magic number and if thats the case the offset returned skips the magic
+   * plus fingerprint bytes.
+   */
+  private def getBinaryOffset(binary: Array[Byte]): Int = {
+    // 2 magic number plus 8 fingerprint bytes
+    val skipOffset = 10;
+    var offset = 0
+    if (binary(0) == 0xC3.toByte && binary(1) == 0x01.toByte) {
+      offset = skipOffset
```

Comment:
It's trivial but I think you can just return 10 here, and return 0 below, for simplicity.